src/contracts/proxies/Vault721.sol

```diff
@@ -256,8 +257,10 @@ contract Vault721 is ERC721EnumerableUpgradeable {
 
   /**
    * @dev check that proxy does not exist OR that the user does not own proxy
+   * revert if a proxy address was input as the _user (EOA or smart contract wallet)
    */
   function _isNotProxy(address _user) internal view returns (bool) {
```

Comment:
- [ ] I think the naming convention for `_isNotProxy` is very confusing. The name implies a check to see if an address is _not a proxy_. But the effective action is to check whether an address owns a proxy. Given that this PR adds an additional reason why a new proxy should not be built, I recommend renaming the function to something like `shouldBuildProxy`
- [ ] I also noticed that you added a revert `IsProxy()` here, yet the behavior of the function is to return a value which may trigger a revert `ProxyAlreadyExist()` in the parent function. My preference, for simplicity would be that either both reverts occur within this function, or that the new lookup `_proxyRegistry` returns the expected value rather than revert.

Reply:
The return boolean indicates whether the proxy needs to be built or not for the user, but if the user is a proxy, it reverts with IsProxy.

Based on the conversation with the auditors, reverting a proxy build to a proxy was the agreed design.